pytorch_lightning/utilities/apply_func.py

```diff
@@ -1,10 +1,14 @@
 from abc import ABC
 from collections import Mapping, Sequence
+from copy import copy
 from typing import Any, Callable, Union
 
 import torch
-from torchtext.data import Batch
-from copy import copy
+
+try:
+    from torchtext.data import Batch
+except ImportError:
+    Batch = None
```

Comment:
```suggestion

import importlib
TORCHTEXT_AVAILABLE = importlib.util.find_spec("torchtext") is not None
```
We need to stop abusing the exception handling please. A package can be present but still throw an ImportError if it was not correctly installed. We need to let the user know that, so we should not catch this exception. Instead, we should let importlib check the package. See #2266

Reply:
well if it is not correctly installed it will crash later, right?

Reply:
Exactly!! so the library will tell the user what was wrong and how to fix it. we dont want to kill this message. This may not apply to torchtext but in general is just good practice, right?